gradle-git-publish 1.0.0 fails on Travis CI when used together with the java plugin

Here is a [sample project](https://github.com/SSCCE/hello-doc) that illustrates the problem.
It generates a simple HTML document and tries to publish it to gh-pages.

The Travis build fails when build.gradle contains the following configuration:
```
plugins {
    id 'java'
    id "org.ajoberstar.git-publish" version '1.0.0'
}
```

However, it succeeds with the following ones:
```
plugins {
    id "org.ajoberstar.git-publish" version '1.0.0'
}
```
and
```
plugins {
    id 'java'
    id "org.ajoberstar.git-publish" version '0.4.1'
}
```


From the command line the gradle build succeeds with all the above configurations.

The Travis logs are available [here](https://travis-ci.org/SSCCE/hello-doc/builds)

I suspect that the new memoization feature is the cause of this issue, but I don't understand why it affects only the Travis build.

**EDIT:** Copying the stack trace from the Travis build:

```
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':gitPublishReset'.
> org.eclipse.jgit.api.errors.TransportException: origin: not found.
* Try:
Run with --debug option to get more log output. Run with --scan to get full insights.
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':gitPublishReset'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
	at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:97)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$ExecuteTaskAction.execute(DefaultTaskExecutionGraph.java:262)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$ExecuteTaskAction.execute(DefaultTaskExecutionGraph.java:246)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:136)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.execute(DefaultTaskPlanExecutor.java:201)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.executeWithTask(DefaultTaskPlanExecutor.java:192)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.internal.UncheckedException: org.eclipse.jgit.api.errors.TransportException: origin: not found.
	at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:63)
	at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:40)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:76)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:794)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:761)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:97)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
	... 30 more
Caused by: org.eclipse.jgit.api.errors.TransportException: origin: not found.
	at org.eclipse.jgit.api.LsRemoteCommand.execute(LsRemoteCommand.java:223)
	at org.eclipse.jgit.api.LsRemoteCommand.call(LsRemoteCommand.java:161)
	at org.eclipse.jgit.api.LsRemoteCommand.call(LsRemoteCommand.java:74)
	at java_util_concurrent_Callable$call$0.call(Unknown Source)
	at org.ajoberstar.grgit.operation.LsRemoteOp.call(LsRemoteOp.groovy:38)
	at org.ajoberstar.grgit.operation.LsRemoteOp.call(LsRemoteOp.groovy)
	at java_util_concurrent_Callable$call.call(Unknown Source)
	at java_util_concurrent_Callable$call.call(Unknown Source)
	at org.ajoberstar.grgit.internal.OpSyntax.consumerOperation(OpSyntax.groovy:25)
	at org.ajoberstar.grgit.internal.OpSyntax$consumerOperation.callStatic(Unknown Source)
	at org.ajoberstar.grgit.Grgit.lsremote(Grgit.groovy)
	at org.ajoberstar.gradle.git.publish.tasks.GitPublishReset.reset(GitPublishReset.java:69)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
	... 43 more
Caused by: org.eclipse.jgit.errors.NoRemoteRepositoryException: origin: not found.
	at org.eclipse.jgit.transport.TransportLocal$1.open(TransportLocal.java:132)
	at org.eclipse.jgit.transport.TransportBundleFile$1.open(TransportBundleFile.java:107)
	at org.eclipse.jgit.transport.Transport.open(Transport.java:570)
	at org.eclipse.jgit.transport.Transport.open(Transport.java:324)
	at org.eclipse.jgit.transport.Transport.open(Transport.java:294)
	at org.eclipse.jgit.api.LsRemoteCommand.execute(LsRemoteCommand.java:188)
	... 55 more
```

--- a/src/compatTest/groovy/org/ajoberstar/gradle/git/publish/BaseCompatTest.groovy
+++ b/src/compatTest/groovy/org/ajoberstar/gradle/git/publish/BaseCompatTest.groovy
@@ -7,6 +7,7 @@ import org.ajoberstar.grgit.Grgit
 import org.gradle.testkit.runner.GradleRunner
 import org.gradle.testkit.runner.BuildResult
 import org.gradle.testkit.runner.TaskOutcome
+import org.gradle.testkit.runner.UnexpectedBuildFailure
 import org.junit.Rule
 import org.junit.rules.TemporaryFolder
 
@@ -239,14 +240,32 @@ gitPublish {
     working.branch.list()*.name == ['gh-pages']
   }
 
+  def 'when no git publish tasks are run, build completes successfully'() {
+    given:
+    buildFile << '''\
+plugins {
+  id 'org.ajoberstar.git-publish'
+}
 
-  private BuildResult build() {
+task hello {
+  doLast {
+    println 'Hello!'
+  }
+}
+'''
+    when:
+    build('hello')
+    then:
+    notThrown(UnexpectedBuildFailure)
+  }
+
+  private BuildResult build(String... args = ['gitPublishPush', '--stacktrace']) {
     return GradleRunner.create()
       .withGradleVersion(System.properties['compat.gradle.version'])
       .withPluginClasspath()
       .withProjectDir(projectDir)
       .forwardOutput()
-      .withArguments('gitPublishPush', '--stacktrace')
+      .withArguments(args)
       .build()
   }
 

--- a/src/main/java/org/ajoberstar/gradle/git/publish/GitPublishPlugin.java
+++ b/src/main/java/org/ajoberstar/gradle/git/publish/GitPublishPlugin.java
@@ -46,7 +46,9 @@ public class GitPublishPlugin implements Plugin<Project> {
     // always close the repo at the end of the build
     project.getGradle().buildFinished(result -> {
       project.getLogger().info("Closing Git publish repo: {}", extension.getRepoDir().get());
-      reset.getGrgit().get().close();
+      if (reset.getGrgit().isPresent()) {
+        reset.getGrgit().get().close();
+      }
     });
   }
 

